modules/services/picom.nix

```diff
@@ -256,33 +224,75 @@ in {
       defaultText = literalExpression "pkgs.picom";
       example = literalExpression "pkgs.picom";
       description = ''
-        picom derivation to use.
+        Picom derivation to use.
       '';
     };
 
-    extraOptions = mkOption {
-      type = types.lines;
-      default = "";
-      example = ''
-        unredir-if-possible = true;
-        dbe = true;
-      '';
-      description = ''
-        Additional Picom configuration.
-      '';
-    };
+    settings = with types;
+      let
+        scalar = oneOf [ bool int float str ] // {
+          description = "scalar types";
+        };
+
+        libConfig = oneOf [ scalar (listOf libConfig) (attrsOf libConfig) ] // {
+          description = "libconfig type";
+        };
+
+        topLevel = attrsOf libConfig // {
+          description = ''
+            libconfig configuration. The format consists of an attributes
+            set (called a group) of settings. Each setting can be a scalar type
+            (boolean, integer, floating point number or string), a list of
+            scalars or a group itself
+          '';
+        };
+
+      in mkOption {
+        type = topLevel;
+        default = { };
+        example = literalExpression ''
+          blur =
+            { method = "gaussian";
+              size = 10;
+              deviation = 5.0;
+            };
```

Comment:
I think this example should be

```suggestion
          {
            blur = {
              method = "gaussian";
              size = 10;
              deviation = 5.0;
            };
          }
```

Otherwise the example will be rendered using invalid Nix syntax:

![picomexample](https://user-images.githubusercontent.com/798147/178456563-167c5078-af2c-43c8-8975-99ae04a8bd5c.png)

But, I noticed that the same error is in the NixOS module so I'll leave this unchanged for now.